src/scripts/main.js

```diff
@@ -1,3 +1,158 @@
 'use strict';
 
 // write code here
+const table = document.querySelector('table');
+const headers = [...table.querySelectorAll('th')];
+const tbody = table.querySelector('tbody');
+const rows = [...tbody.querySelectorAll('tr')];
+let lastSortedColumn = null;
+let isDescending = false;
+
+const getNumber = (n) => Number(n.replace(/[^0-9.-]+/g, ''));
+
+headers.forEach((header, i) => {
+  header.addEventListener('click', () => {
+    if (lastSortedColumn === i) {
+      isDescending = !isDescending;
+    } else {
+      lastSortedColumn = i;
+      isDescending = false;
+    }
+
+    rows.sort((a, b) => {
+      const cellA = a.querySelectorAll('td')[i].textContent;
+      const cellB = b.querySelectorAll('td')[i].textContent;
+
+      if (Number(getNumber(cellA))) {
+        return isDescending
+          ? getNumber(cellB) - getNumber(cellA)
+          : getNumber(cellA) - getNumber(cellB);
+      } else {
+        return isDescending
+          ? cellB.localeCompare(cellA)
+          : cellA.localeCompare(cellB);
+      }
+    });
+
+    rows.forEach((row) => tbody.appendChild(row));
+  });
+});
+
+table.addEventListener('click', (e) => {
+  const row = e.target.closest('tr');
+
+  if (row) {
+    table.querySelectorAll('tr').forEach((i) => i.classList.remove('active'));
+    row.classList.add('active');
+  }
+});
+
+const forma = document.createElement('form');
+
+forma.className = 'new-employee-form';
+
+forma.innerHTML = `
+  <label>Name: <input name="name" type="text" data-qa="name" required></label>
+  <label>Position: <input name="position" type="text" data-qa="position" required></label>
+  <label>Office:
+    <select name="office" data-qa="office" required>
+      <option>Tokyo</option>
+      <option>Singapore</option>
+      <option>London</option>
+      <option>New York</option>
+      <option>Edinburgh</option>
+      <option>San Francisco</option>
+    </select>
+  </label>
+  <label>Age: <input name="age" type="number" data-qa="age" required></label>
+  <label>Salary: <input name="salary" type="number" data-qa="salary" required></label>
+  <button data-qa="save-button">Save to table</button>
+`;
+
+document.body.append(forma);
+
+const form = document.forms[0];
+const button = form.querySelector('button[data-qa="save-button"]');
+
+const getFormData = () => ({
+  name: form.elements.name.value.trim(),
+  position: form.elements.position.value.trim(),
+  office: form.elements.office.value.trim(),
+  age: parseInt(form.elements.age.value, 10),
+  salary: parseFloat(form.elements.salary.value.replace(/[^0-9.]/g, '')),
```

Comment:
The salary parsing logic might not handle all currency formats correctly. Ensure that the regex used in `replace` method removes all non-numeric characters except the decimal point.